src/main/java/jenkins/security/plugins/ldap/LDAPConfiguration.java

```diff
@@ -405,18 +412,37 @@ public FormValidation doCheckServer(@QueryParameter String value, @QueryParamete
 
             try {
                 Hashtable<String,String> props = new Hashtable<String,String>();
-                if(managerDN!=null && managerDN.trim().length() > 0  && !"undefined".equals(managerDN)) {
-                    props.put(Context.SECURITY_PRINCIPAL,managerDN);
+                props.put(Context.INITIAL_CONTEXT_FACTORY, "com.sun.jndi.ldap.LdapCtxFactory");
+                props.put(Context.PROVIDER_URL, LDAPSecurityRealm.toProviderUrl(server,rootDN));
+
+                LdapContext ctx = new InitialLdapContext(props, null);
+                StartTlsResponse tls = (StartTlsResponse) ctx.extendedOperation(new StartTlsRequest());
+                SSLSession session = tls.negotiate();
+
+                if (!session.isValid()) {
```

Comment:
does this not break for anyone using non SSL LDAP? (in other words this should be available for users to opt into, or opt out of).
For the AD plugin we also need to close the `StartTlsResponse` https://github.com/jenkinsci/active-directory-plugin/blob/19e9fbcf0518537c6c4bd24fb1fd09901e7fd60e/src/main/java/hudson/plugins/active_directory/ActiveDirectorySecurityRealm.java#L627-L644

Reply:
agree definitely using startTLS must be optional and not per default.

Reply:
agreed, this definitely breaks non-STARTTLS implementations. I commented below with some additional context for this PR.